### PR TITLE
Remove IndentingTextWriter in favor of the existing IndentedTextWriter(#1241)

### LINKED ISSUE #1241: Remove IndentingTextWriter in favor of the existing IndentedTextWriter
Remove ML.NET's IndentingTextWriter in favor of System.CodeDom.Compiler.IndentedTextWriter (a mainstream type in S.Runtime.Extensions.dll just in an oddly chosen namespace)

The main thing IndentedTextWriter doesn't have is the little IDisposable Scope helper that ML.NET uses in a decent number of places, that lets you write code like `using (writer.Nest()) { /* indented */ }` rather than explicitly indenting and outdenting before and after.  For now it could be left as an 8-line helper in ML.NET, but based on IndentedTextWriter instead of their IndentingTextWriter, and optionally open an API proposal in the CoreFX repo to add such a small feature.

Per discussions w/ @stephentoub @Zruty0

### PR DESCRIPTION
**Pull request by feiyun0112**
Fix #1241
Use System.CodeDom.Compiler.IndentedTextWriter

### CODE REVIEW

**Comment by Ivanidzo4ka** [comment]
Thank you for your contribution @feiyun0112! Can you run, let's say SavePipeCat test, on your local machine and find out why baselines are not longer match?

**Comment by Ivanidzo4ka** [comment]
You almost there! Last thing which is failing is TestGeneratedCSharpAPI. In order to unblock yourself, you need to find RegenerateCSharpApitest, remove skip from it, run it locally and them checkin changes in CSharpApi.cs

File: src/Microsoft.ML.Core/Utilities/IndentedTextWriterExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                _wrt.Indent();
+            }
+            public void Dispose()
+            {
+                _wrt.Outdent();
+                _wrt = null;
+            }
+        }
+
+        public static Scope Nest(this IndentedTextWriter wrt)
```
Comment (eerhardt) [inline src/Microsoft.ML.Core/Utilities/IndentedTextWriterExtensions.cs]: The parameter names should be full words - `writer`

File: src/Microsoft.ML.Core/Utilities/IndentedTextWriterExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.CodeDom.Compiler;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.ML.Runtime.Internal.Utilities
+{
+    public static class IndentedTextWriterExtensions
```
Comment (eerhardt) [inline src/Microsoft.ML.Core/Utilities/IndentedTextWriterExtensions.cs]: Can this be internal and have a `[BestFriend]` attribute on it? I don't believe we want to ship this API publicly in our core assembly.

File: src/Microsoft.ML.Core/Utilities/IndentedTextWriterExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            return new Scope(wrt);
+        }
+
+        public static void Indent(this IndentedTextWriter wrt)
+        {
+            wrt.Indent++;
+        }
+        public static void Outdent(this IndentedTextWriter wrt)
+        {
+            --wrt.Indent;
```
Comment (eerhardt) [inline src/Microsoft.ML.Core/Utilities/IndentedTextWriterExtensions.cs]: The `--` can go after `wrt.Indent`. It doesn't matter either way since we aren't capturing the return value, but putting it after makes it consistent with the `++` in the method above.

File: src/Microsoft.ML.Core/Microsoft.ML.Core.csproj [hunk 4]
```diff
@@ -12,6 +12,7 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="System.CodeDom" Version="$(SystemCodeDomPackageVersion)" />
```
Comment (eerhardt) [inline src/Microsoft.ML.Core/Microsoft.ML.Core.csproj]: Is this necessary?  I don't believe it is. The `IndentedTextWriter` should be in the `netstandard` assembly, and no extra packagereference is needed.

File: src/Microsoft.ML.Legacy/Runtime/Internal/Tools/CSharpApiGenerator.cs [hunk 5]
```diff
@@ -115,10 +116,10 @@ private void GenerateInputOutput(IndentingTextWriter writer, ComponentCatalog.En
             GenerateInput(writer, entryPointInfo, catalog);
             writer.Outdent();
             writer.WriteLine("}");
-            writer.WriteLine();
+            writer.WriteLineNoTabs();
```
Comment (eerhardt) [inline src/Microsoft.ML.Legacy/Runtime/Internal/Tools/CSharpApiGenerator.cs]: Why is this change necessary?  here and below

Comment (feiyun0112) [reply src/Microsoft.ML.Legacy/Runtime/Internal/Tools/CSharpApiGenerator.cs]: Yes, it's necessary.

`writer.WriteLine();` will output tabs

**Comment by Zruty0** [comment]
@feiyun0112 could you please merge with master?

**Comment by Ivanidzo4ka** [comment]
Thank you for doing this @feiyun0112 !
